auth-api/src/auth_api/models/user.py

```diff
@@ -42,16 +43,18 @@ class User(db.Model):
     modified = Column(DateTime)
     roles = Column('roles', String(1000))
 
+    contacts = relationship('ContactLink', back_populates='user')
+
     @classmethod
-    def find_by_keycloak_guid(cls, keycloak_guid):
+    def find_by_username(cls, username):
         """Return the first user with the provided KeyCloak GUID."""
-        return cls.query.filter_by(keycloak_guid=keycloak_guid).first()
+        return cls.query.filter_by(username=username).first()
 
     @classmethod
     def find_by_jwt_token(cls, token: dict):
-        """Return if they exist and match the provided JWT."""
+        """Find an existing user by the keycloak GUID in the provided token."""
         return cls.query.filter_by(
-            username=token.get('preferred_username', None)
+            keycloak_guid=token.get('sub', None)
```

Comment:
just to bring the abstraction , may be pass only the value of token.get('sub', None) instead of whole token,... i am not sure which one makes sense

Reply:
Yes, you're right.

Reply:
Actually, I want to keep this one the way it is.  The reason is that if later on we decide to change the field in the token that we use to fetch the user, we'll only have to change it in one place in the model (`find_by_jwt_token`).  If we were to change it to `find_by_keycloak_guid`, we'd have to change it in several places in the service.